src/main/cpp/DragonVision/DragonCamera.h

```diff
@@ -0,0 +1,118 @@
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+// OR OTHER DEALINGS IN THE SOFTWARE.
+//====================================================================================================================================================
+#pragma once
+
+#include "units/angle.h"
+#include "units/length.h"
+#include "units/time.h"
+
+#include <string>
+#include <vector>
+#include "frc/geometry/Pose3d.h"
+#include "frc/DriverStation.h"
+#include "DragonVision/DragonVisonStructs.h"
+
+class DragonCamera
+{
+public:
+    enum PIPELINE
+    {
+        OFF,
+        MACHINE_LEARNING,
+        APIRL_TAG,
+        COLOR_THRESHOLD
+    };
+
+    DragonCamera(
+        std::string cameraName,                /// <I> camera name/type
+        PIPELINE pipeline,                     /// <I> enum for pipeline
+        units::length::inch_t mountingXOffset, /// <I> x offset of cam from robot center (forward relative to robot is positive)
+        units::length::inch_t mountingYOffset, /// <I> y offset of cam from robot center (left relative to robot is positive)
+        units::length::inch_t mountingZOffset, /// <I> z offset of cam from robot center (up relative to robot is positive)
+        units::angle::degree_t pitch,          /// <I> - Pitch of camera
+        units::angle::degree_t yaw,            /// <I> - Yaw of camera
+        units::angle::degree_t roll            /// <I> - Roll of camera
+    );
+    DragonCamera() = delete;
+
+    virtual bool HasTarget() const = 0;
+
+    // Getters
+    virtual units::angle::degree_t GetTargetHorizontalOffset() const = 0;
+    virtual units::angle::degree_t GetTargetHorizontalOffsetRobotFrame(units::length::inch_t *targetDistOffset_RF, units::length::inch_t *targetDistfromRobot_RF) const = 0;
+    virtual units::angle::degree_t GetTargetVerticalOffsetRobotFrame(units::length::inch_t *targetDistOffset_RF, units::length::inch_t *targetDistfromRobot_RF) const = 0;
+    virtual units::angle::degree_t GetTargetVerticalOffset() const = 0;
+    virtual units::time::microsecond_t GetPipelineLatency() const = 0;
+    virtual units::angle::degree_t GetTargetSkew() const = 0;
+    virtual double GetTargetArea() const = 0;
+    virtual int GetAprilTagID() const = 0;
+
+    virtual VisionPose GetFieldPosition() const = 0;
+    virtual VisionPose GetFieldPosition(frc::DriverStation::Alliance alliance) const = 0;
+
+    //  Estimating distance
+    virtual units::length::inch_t EstimateTargetXDistance() const = 0;
+    virtual units::length::inch_t EstimateTargetYDistance() const = 0;
+    virtual units::length::inch_t EstimateTargetZDistance() const = 0;
+
+    virtual units::length::inch_t EstimateTargetXDistance_RelToRobotCoords() const = 0;
+    virtual units::length::inch_t EstimateTargetYDistance_RelToRobotCoords() const = 0;
+    virtual units::length::inch_t EstimateTargetZDistance_RelToRobotCoords() const = 0;
+
```

Comment:
Should these function names start with "get"?
    //  Estimating distance
    virtual units::length::inch_t EstimateTargetXDistance() const = 0;
    virtual units::length::inch_t EstimateTargetYDistance() const = 0;
    virtual units::length::inch_t EstimateTargetZDistance() const = 0;

    virtual units::length::inch_t EstimateTargetXDistance_RelToRobotCoords() const = 0;
    virtual units::length::inch_t EstimateTargetYDistance_RelToRobotCoords() const = 0;
    virtual units::length::inch_t EstimateTargetZDistance_RelToRobotCoords() const = 0;

Reply:
Fixed.